NachoSpigot-Server/src/main/java/me/rastrian/dev/threads/netty/Tasks.java

```diff
@@ -0,0 +1,33 @@
+package me.rastrian.dev.threads.netty;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class Tasks {
+    private final AtomicInteger state = new AtomicInteger();
+
+    public boolean fetchTask() {
+        int old = this.state.getAndDecrement();
+        if (old == State.RUNNING_GOT_TASKS.ordinal()) {
+            return true;
+        }
+        if (old == State.RUNNING_NO_TASKS.ordinal()) {
+            return false;
+        }
+        throw new AssertionError();
+    }
+
+    public boolean addTask() {
+        if (this.state.get() == State.RUNNING_GOT_TASKS.ordinal()) {
+            return false;
+        }
+        int old = this.state.getAndSet(State.RUNNING_GOT_TASKS.ordinal());
+        return old == State.WAITING.ordinal();
+    }
+
+    private static enum State {
```

Comment:
```suggestion
    private enum State {
```
Static is implicit. [Documentation](https://docs.oracle.com/javase/specs/jls/se7/html/jls-8.html#jls-8.9)
> Nested enum types are implicitly static.